fiftyone/brain/internal/core/leaky_splits.py

```diff
@@ -1,108 +1,140 @@
 
     config = LeakySplitsConfig(
-        split_views,
-        split_field,
-        split_tags,
-        similarity_brain_key,
-        embeddings_field,
-        model,
-        model_kwargs,
-        similarity_backend,
-        similarity_config_dict,
-        **kwargs,
+        split_field=split_field,
+        split_tags=split_tags,
+        embeddings_field=embeddings_field,
+        similarity_index=similarity_index,
+        model=model,
+        model_kwargs=model_kwargs,
     )
+
     brain_method = config.build()
     brain_method.ensure_requirements()
 
-    if brain_key is not None:
-        # Don't allow overwriting an existing run with same key, since we
-        # need the existing run in order to perform workflows like
-        # automatically cleaning up the backend's index
-        brain_method.register_run(samples, brain_key, overwrite=False)
-
-    results = brain_method.initialize(samples, brain_key)
-
-    if results._save_similarity_index:
-        results._similarity_method.register_run(
+    if similarity_index is None:
+        similarity_index = fb.compute_similarity(
             samples,
-            results.config.similarity_brain_key,
+            backend="sklearn",
+            model=model,
+            model_kwargs=model_kwargs,
+            embeddings=embeddings_field or embeddings,
+            batch_size=batch_size,
+            num_workers=num_workers,
+            skip_failures=skip_failures,
+            progress=progress,
         )
-
-    results.set_threshold(threshold)
-    leaks = results.leaks_view()
-
-    if brain_key is not None:
-        brain_method.save_run_results(samples, brain_key, results)
-    if results._save_similarity_index:
-        results._similarity_method.save_run_results(
-            samples, similarity_brain_key, results._similarity_index
+    elif not isinstance(similarity_index, fbs.DuplicatesMixin):
+        raise ValueError(
+            "This method only supports similarity indexes that implement the "
+            "%s mixin" % fbs.DuplicatesMixin
         )
 
-    return results, leaks
+    split_views = _to_views(
+        samples,
+        split_tags=split_tags,
+        split_field=split_field,
+        split_views=split_views,
+    )
+
+    index = brain_method.initialize(samples, similarity_index, split_views)
 
+    if threshold is not None:
+        index.find_leaks(threshold=threshold)
 
-### GENERAL
+    return index
 
 
 class LeakySplitsConfig(fob.BrainMethodConfig):
     def __init__(
         self,
-        split_views=None,
         split_field=None,
         split_tags=None,
-        similarity_brain_key=None,
         embeddings_field=None,
+        similarity_index=None,
         model=None,
         model_kwargs=None,
-        similarity_backend=None,
-        similarity_config_dict=None,
         **kwargs,
     ):
-        self.split_views = split_views
+        if similarity_index is not None and not etau.is_str(similarity_index):
```

Comment:
All public properties of `BrainMethodConfig` classes must be JSON serializable, as this classes' public properties are automatically serialized and made available  via `load_brain_info().config` (**IF** the brain run is stored under a `brain_key`)